conan/tools/premake/premake.py

```diff
@@ -1,40 +1,96 @@
+import textwrap
+from pathlib import Path
+
+from jinja2 import Template
+
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import save
+from conan.tools.microsoft.msbuild import MSBuild
+from conan.tools.premake.toolchain import PremakeToolchain
+from conan.tools.premake.constants import CONAN_TO_PREMAKE_ARCH
+
 # Source: https://learn.microsoft.com/en-us/cpp/overview/compiler-versions?view=msvc-170
 PREMAKE_VS_VERSION = {
-    '190': '2015',
-    '191': '2017',
-    '192': '2019',
-    '193': '2022',
-    '194': '2022',  # still 2022
+    "190": "2015",
+    "191": "2017",
+    "192": "2019",
+    "193": "2022",
+    "194": "2022",  # still 2022
 }
 
-
 class Premake:
     """
     Premake cli wrapper
     """
 
+    filename = "conanfile.premake5.lua"
+
+    # Conan premake file which will preconfigure toolchain and then will call the user's premake file
+    _premake_file_template = textwrap.dedent(
+        """\
+    #!lua
+    include("{{luafile}}")
+    include("{{premake_conan_toolchain}}")
+    """
+    )
+
     def __init__(self, conanfile):
         self._conanfile = conanfile
-
-        self.action = None              # premake5 action to use (Autogenerated)
-        self.luafile = 'premake5.lua'   # Path to the root (premake5) lua file
+        # Path to the root (premake5) lua file
+        self.luafile = (Path(self._conanfile.source_folder) / "premake5.lua").as_posix()
+        print("luafile", self.luafile)
         # (key value pairs. Will translate to "--{key}={value}")
-        self.arguments = {}             # https://premake.github.io/docs/Command-Line-Arguments/
+        self.arguments = {}  # https://premake.github.io/docs/Command-Line-Arguments/
+        # self.arguments["scripts"] = Path(self._conanfile.generators_folder).as_posix()
+
+        arch = str(self._conanfile.settings.arch)
+        if arch not in CONAN_TO_PREMAKE_ARCH:
+            raise ConanInvalidConfiguration(
```

Comment:
ConanInvalidConfiguration is not intended for this point, only for ``validate()``